app-modules/service-management/tests/ServiceRequest/ListServiceRequestsTest.php

```diff
@@ -137,3 +140,51 @@
             ServiceRequestResource::getUrl()
         )->assertSuccessful();
 });
+
+test('can filter service request by organization', function () {
+    asSuperAdmin();
+
+    $organization = Organization::factory()->create();
+
+    $contacts = Contact::factory()
+        ->count(10)
+        ->state(['organization_id' => $organization->id])
+        ->create();
+
+    $serviceRequests = ServiceRequest::factory()
+        ->count(10)
+        ->create()
+        ->each(function ($serviceRequest) use ($contacts) {
+            $serviceRequest->respondent()->associate($contacts->random())->save();
+        });
+
+    $organization = $serviceRequests->first()->respondent->organizations;
+
+    $organizationNotInList = Organization::factory()->create();
+
+    $filterdServiceRequests = ServiceRequest::whereHasMorph(
+        'respondent',
+        [Contact::class],
+        fn (Builder $query): Builder => $query->whereRelation(
+            'organizations',
+            (new Organization())->getKeyName(),
+            $organization->id
+        )
+    )->get();
+
+    $unfilterdServiceRequests = ServiceRequest::whereHasMorph(
+        'respondent',
+        [Contact::class],
+        fn (Builder $query): Builder => $query->whereRelation(
+            'organizations',
+            (new Organization())->getKeyName(),
+            $organizationNotInList->id
+        )
+    )->get();
```

Comment:
I would not recommend writing the same or similar code in your tests as what you are testing in the code itself.

Statically define the contacts and organizations you would expect to see displayed in the list and then assert that the code written changes the list correctly.

If you have the same code in your test as you do in the code it is testing, then it slightly defeats the purpose of the assertion.